src/sumInput.js

```diff
@@ -0,0 +1,22 @@
+
+const sumInput = (parseNum) => {
+    let sum = 0;
+
+    // forEach() 메서드로 우선 입력된 값중에 너무 큰 값이 있는지 검사하고 있다면 Error 반환
+    // 통과하면 배열의 모든 수를 합산
+    parseNum.forEach(value => {
+        if (value > Number.MAX_SAFE_INTEGER) {
+            throw new Error('[ERROR] 입력된 값들 중 너무 큰 값이 있습니다.');
+        }
+        sum += value;
+    });
+
+    // 마지막으로 합산된 값이 infinity를 출력할 정도로 너무 크지 않은지 검사
+    if (sum > Number.MAX_SAFE_INTEGER) {
+        throw new Error('[ERROR] 합산된 값이 너무 큽니다.')
+    }
+
+    return sum;
```

Comment:
정말 꼼꼼하게 에러 처리를 잘 하신 것 같아요 ㅎㅎㅎ 이 케이스는 생각지도 못했네요 ㅎㅎ👍👍
지금도 너무 좋지만 이렇게도 변경해 볼 수 있을 것 같아요! 👍
```suggestion
    // 우선 입력된 값중에 너무 큰 값이 있는지 검사하고 있다면 Error 반환
    const numberIsTooBig = parseNum.some((value)=> value > Number.MAX_SAFE_INTEGER)
    if(numberIsTooBig) throw new Error('[ERROR] 입력된 값들 중 너무 큰 값이 있습니다.');
    
    // 통과하면 배열의 모든 수를 합산
    const sum = parseNum.reduce((a,b)=>a+b,0)

    // 마지막으로 합산된 값이 infinity를 출력할 정도로 너무 크지 않은지 검사
    if (sum > Number.MAX_SAFE_INTEGER) {
        throw new Error('[ERROR] 합산된 값이 너무 큽니다.')
    }

    return sum;
```

Reply:
우와 깃허브에 이렇게 코드 제안 리뷰 기능이 있네요?!👍
맞아요 reduce()함수를 썼으면 간단하게 처리할 수 있었는데 좀 돌아갔어요 ㅠㅠㅎㅎ
 const numberIsTooBig = parseNum.some((value)=> value > Number.MAX_SAFE_INTEGER)
    if(numberIsTooBig) throw new Error('[ERROR] 입력된 값들 중 너무 큰 값이 있습니다.');
이 부분도 some()메소드로 이렇게 처리할 수 있었겠네요!! 좋은 코드 제안 감사합니다!☺️

Reply:
<img width="388" alt="스크린샷 2024-10-23 오후 11 26 22" src="https://github.com/user-attachments/assets/d4deeef0-fff1-4964-a513-a29413d82522">
suggestion 누르면 코드 제안을 할 수 있습니다!! 👍